subt/tools/analyze_detectors.py

```diff
@@ -0,0 +1,243 @@
+"""evaluate detectors"""
+import os
+import numpy as np
+import cv2
+from matplotlib import pyplot as plt
+
+try:
+    import torch
+    import subt.artf_model
+    from subt.artf_detector import Detector
+except ImportError:
+    print('\nWarning: missing torch!\n')
+
+from osgar.logger import LogReader, lookup_stream_id
+from osgar.lib.serialize import deserialize
+from subt.tf_detector import CvDetector
+from subt.artf_node import check_results
+
+g_streams = ["logimage.image", "logimage_rear.image"]
+g_artf_names = ["survivor","backpack", "phone", "helmet", "rope", "fire_extinguisher", "drill", "vent", "cube"]
+
+NAMES_AND_SCORES = {'backpack': 0.1,
+                    'survivor': 0.1,
+                    'phone': 0.1,
+                    'rope': 0.1,
+                    'helmet': 0.1,
+                    'fire_extinguisher': 0.1,
+                    'drill': 0.1,
+                    'vent': 0.1,
+                    'cube': 0.1,
+                    'robot': 1.0,
+                    'breadcrumb': 1.0,
+                    'nothing': 1.0
+                    }
+MIN_SCORES = np.array(list(NAMES_AND_SCORES.values()))
+
+
+def take_third(item):
+    return item[2]
+
+
+def proces_torch_result(result):
+    result.sort(key=take_third, reverse=True)
```

Comment:
My mild preference would be to use a lambda function instead of creating the `take_third` function. I.e. `result.sort(key=lambda item: item[2], reverse=True)`. I will not insist though :-)

Reply:
I am not much familiar with lambda, but it looks clearer.